src/sumoLogger.js

```diff
@@ -73,8 +82,25 @@ class SumoLogger {
         }
     }
 
+    batchReadyToSend() {
+        if (this.config.batchSize === 0) {
+            return this.config.interval === 0;
+        } else {
+            const pendingMessages = this.pendingLogs.reduce((acc, curr) => {
+                const log = JSON.parse(curr);
+                return acc + log.msg + '\n';
+            }, '');
+            const pendingBatchSize = pendingMessages.length;
+            const ready = pendingBatchSize >= this.config.batchSize;
+            if (ready) {
+                this.stopLogSending();
+            }
+            return ready;
+        }
+    }
+
     sendLogs() {
-        if (this.pendingLogs.length === 0) {
+        if (this.pendingLogs .length === 0) {
```

Comment:
Incorrect extra space added before `.length`

Reply:
fixed

Reply:
@clementallen can you review changes? Thanks!